pkg/controller/testsuite/testsuite_controller.go

```diff
@@ -87,26 +92,34 @@ var _ reconcile.Reconciler = &ReconcileTestSuite{}
 // ReconcileTestSuite reconciles a ClusterTestSuite object
 type ReconcileTestSuite struct {
 	client.Client
-	scheme *runtime.Scheme
+	scheme            *runtime.Scheme
+	scheduler         TestScheduler
+	reporter          TestReporter
+	statusService     SuiteStatusService
+	definitionService TestDefinitionService
 }
 
+const (
+	defaultRequeueAfter = time.Second * 10
+)
+
 // Reconcile reads that state of the cluster for a ClusterTestSuite object and makes changes based on the state read
 // and what is in the ClusterTestSuite.Spec
-// TODO(user): Modify this Reconcile function to implement your Controller logic.  The scaffolding writes
-// a Deployment as an example
+
 // Automatically generate RBAC rules to allow the Controller to read and write Deployments
 // +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=apps,resources=deployments/status,verbs=get;update;patch
 // +kubebuilder:rbac:groups=testing.kyma-project.io,resources=testsuites,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=testing.kyma-project.io,resources=testsuites/status,verbs=get;update;patch
 func (r *ReconcileTestSuite) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	// Fetch the ClusterTestSuite instance
-	instance := &testingv1alpha1.ClusterTestSuite{}
+	ctx := context.Background()
+	// Fetch the ClusterTestSuite suiteCopy
+	suite := &testingv1alpha1.ClusterTestSuite{}
 	// TODO quick fix for cluster-scoped objects
 	request.Namespace = ""
-	err := r.Get(context.TODO(), request.NamespacedName, instance)
+	err := r.Get(context.TODO(), request.NamespacedName, suite)
 	if err != nil {
-		if errors.IsNotFound(err) {
+		if k8serrors.IsNotFound(err) {
 			// Object not found, return.  Created objects are automatically garbage collected.
```

Comment:
That's not true yet (TODO?). scheduler is not setting ownerReference and pods are not deleted after test suite is removed.

Reply:
fixed